odoo/api.py

```diff
@@ -558,7 +559,8 @@ def company(self):
                 user_company_ids = self.user.company_ids.ids
                 if any(cid not in user_company_ids for cid in company_ids):
                     raise AccessError(_("Access to unauthorized or invalid companies."))
-            return self['res.company'].browse(company_ids[0])
+            # sudoed environment to be coherent with the user property logic
+            return self['res.company'].sudo().browse(company_ids[0])
```

Comment:
```suggestion
            return self(su=True)['res.company'].browse(company_ids[0])
```
actually saves one browse() :wink:

Reply:
And it is consistent with the code of `user`.